MinGW compile-check fails: displayxr_mcp pthreads check runs before xrt-pthreads target is defined

## Symptom

`./scripts/build-mingw-check.sh` (cross-compile sanity check from macOS / Linux) fails at the configure step with:

```
CMake Error at build-mingw/_deps/displayxr_mcp-src/CMakeLists.txt:87 (message):
  displayxr_mcp on Windows requires vcpkg 'pthreads' (or the runtime's
  xrt-pthreads target).
```

## Root cause

In the runtime's top-level `CMakeLists.txt`:

- Line 119–129: `FetchContent_MakeAvailable(displayxr_mcp)` runs.
- Line 171–180: `add_library(xrt-pthreads INTERFACE)` defines the target.

`displayxr_mcp/CMakeLists.txt:87` does `if(TARGET xrt-pthreads)` first, then falls back to `find_package(PThreads4W CONFIG QUIET)`. On a MinGW cross-compile, neither path satisfies (xrt-pthreads not yet defined; PThreads4W not present — vcpkg isn't on the toolchain), so the FATAL_ERROR fires.

This is order-dependent — for MSVC builds it doesn't matter because `find_package(PThreads4W ...)` succeeds via vcpkg later in the FetchContent's own search. For MinGW it does matter because the toolchain has neither, but it does have winpthreads via `find_package(Threads)`.

## Fix options

**Option A — local (this repo).** Move the `add_library(xrt-pthreads INTERFACE)` block (lines 171–180 of root `CMakeLists.txt`) to *before* the `FetchContent_MakeAvailable(displayxr_mcp)` call at line 129. Then `displayxr_mcp/CMakeLists.txt:87` finds `xrt-pthreads` via its first branch on every platform, including MinGW.

Risk: MSVC builds would now hit the `xrt-pthreads` branch instead of the `PThreads4W::PThreads4W` branch in displayxr_mcp's check. Since `xrt-pthreads` already wraps `PThreads4W::PThreads4W_CXXEXC` for MSVC (line 174), the resulting link target chain is functionally equivalent. Worth verifying in CI before merge.

**Option B — upstream (`DisplayXR/displayxr-mcp`).** Relax `displayxr-mcp/CMakeLists.txt:87` to fall through to `find_package(Threads)` on MinGW (`if(MINGW) ... Threads::Threads`). Cleaner long-term — handles other MinGW consumers — but requires a release of `displayxr-mcp` and a `GIT_TAG` bump here.

Recommendation: Option A. Two-line move; immediate; no upstream coordination.

## Workaround for now

Skip MinGW for changes scoped to MSVC-only TUs (`comp_d3d11_service.cpp` etc., per CLAUDE.md). Real Windows validation lands via CI on PR.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,6 +126,24 @@ FetchContent_Declare(displayxr_mcp
 # instead, so suppress install rules here.
 set(DISPLAYXR_MCP_BUILD_ADAPTER ON CACHE BOOL "" FORCE)
 set(DISPLAYXR_MCP_INSTALL OFF CACHE BOOL "" FORCE)
+
+# xrt-pthreads must exist before FetchContent_MakeAvailable processes
+# displayxr_mcp's CMakeLists.txt — its check at line 87 picks up the
+# parent project's `xrt-pthreads` target via the first branch when
+# present, otherwise falls back to vcpkg `PThreads4W`. Without this
+# ordering, MinGW cross-compile fails configure (no vcpkg, no
+# xrt-pthreads yet defined). See #208.
+add_library(xrt-pthreads INTERFACE)
+if(WIN32 AND NOT MINGW)
+	find_package(pthreads_windows REQUIRED)
+	target_link_libraries(xrt-pthreads INTERFACE PThreads4W::PThreads4W_CXXEXC)
+else()
+	set(CMAKE_THREAD_PREFER_PTHREAD ON)
+	find_package(Threads)
+	target_link_libraries(xrt-pthreads INTERFACE Threads::Threads)
+	target_compile_definitions(xrt-pthreads INTERFACE _GNU_SOURCE)
+endif()
+
 FetchContent_MakeAvailable(displayxr_mcp)
 
 if(NOT WIN32)
@@ -168,16 +186,9 @@ endif()
 #https://github.com/arsenm/sanitizers-cmake
 find_package(Sanitizers MODULE)
 
-add_library(xrt-pthreads INTERFACE)
-if(WIN32 AND NOT MINGW)
-	find_package(pthreads_windows REQUIRED)
-	target_link_libraries(xrt-pthreads INTERFACE PThreads4W::PThreads4W_CXXEXC)
-else()
-	set(CMAKE_THREAD_PREFER_PTHREAD ON)
-	find_package(Threads)
-	target_link_libraries(xrt-pthreads INTERFACE Threads::Threads)
-	target_compile_definitions(xrt-pthreads INTERFACE _GNU_SOURCE)
-endif()
+# xrt-pthreads is now defined above, just before
+# FetchContent_MakeAvailable(displayxr_mcp), so its target check
+# resolves on every platform including MinGW. See #208.
 
 find_package(OpenGL)
 set(OPENGL_WITHOUT_GLX_FOUND ${OPENGL_FOUND})

